src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,138 @@
+import {
+    Component,
+    Input,
+    ViewEncapsulation,
+    ContentChildren,
+    QueryList,
+    Renderer2,
+    ElementRef,
+    Optional,
+    SkipSelf,
+    ContentChild,
+    Output,
+    AfterViewInit,
+  HostListener
+} from '@angular/core';
+import { Router, RouterLinkActive, NavigationEnd, RouterState, ActivatedRoute, UrlTree } from '@angular/router';
+import 'rxjs/add/operator/filter';
+
+@Component({
+    selector: '[ux-navigation-item]',
+    templateUrl: './navigation-item.component.html',
+    host: {
+        '[class.active]': 'active',
+        '[class.selected]': 'expanded',
+    },
+    encapsulation: ViewEncapsulation.None
+})
+export class NavigationItemComponent implements AfterViewInit {
+    @Input() header: string;
+    @Input() icon: string;
+    @Input() expanded: boolean = false;
+    @Input() link: string;
+
+    @Output()
+    get level(): number {
+        return this._level;
+    }
+
+    @Output()
+    get active() {
+        if (this.link) {
+            return this._router.isActive(this.link, true);
+        }
+    }
+
+    private _level: number = 1;
+    private _active: boolean = false;
+
+    @ContentChildren(NavigationItemComponent, { descendants: true })
+    private _children: QueryList<NavigationItemComponent>;
+
+    get children(): NavigationItemComponent[] {
+        return this._children.filter(item => item !== this);
+    }    
+
+    constructor(
+        private _elementRef: ElementRef,
+        private _renderer: Renderer2,
+        @Optional()
+        @SkipSelf()
+        private _parent: NavigationItemComponent,
+        @Optional() private _router: Router
+    ) {
+        this._level = _parent ? _parent.level + 1 : 1;
+
+        _router.events.filter(event => event instanceof NavigationEnd).subscribe(event => {              
+            if (this.link && this._router.isActive(this.link, true)) {
+                this.select();
+            }
+
+            this.expanded = this.hasActiveLink(this.link);
+        });
+    }
+
+    ngAfterViewInit(): void {
+        // Add class to parent for styling
+        const parentListElement = this._elementRef.nativeElement.parentElement;
+        if (parentListElement) {
+            let levelClass: string = this.getLevelClass();
+            if (levelClass.length > 0) {
+                this._renderer.addClass(parentListElement, levelClass);
+            }
+        }
+    }
+
+    toggle(): void {
```

Comment:
Remove this function - this can be done in the view which is more performant, eg:

```
(click)="expanded = !expanded"
```

Reply:
Function removed and view updated as suggested.